### PR TITLE
FEATURE: add `--raw`. `--tabs` and `--indent` to `to nuon` as in `to json`

### LINKED ISSUE #7255: Add whitespace pretty-printing to `to nuon`
### Related problem

_No response_

### Describe the solution you'd like

Very simply: `to json` pretty-prints the output by default, so it makes sense that `to nuon` also should. (It can have a `-r` `--raw` option added to re-enable the current compressed output).

Note that the default whitespace formatting will have to be implemented in-house. Also, it'd probably be expected to implement `to json`'s `-i` and `-t` switches.

### Describe alternatives you've considered

_No response_

### Additional context and details

_No response_

### PR DESCRIPTION
**Pull request by amtoine**
Should close #7255.

# Description
**TL;DR**: this PR adds `--indent <int>`, `--tabs <int>` and `--raw` to control a bit more the `string` output of `to nuon`, as done in `to json` already, the goal being to promote the `NUON` format through easy to read and formatted output `.nuon` files :yum: 

### outside of `crates/nu-command/src/formats/to/nuon.rs`
as the signature of `value_to_string` has changed, the single call to it outside of its module definition has been changed to use default values => `value_to_string(&value, Span::unknown(), 0, &None)` in `crates/nu-command/src/filters/uniq.rs`

### changes to `ToNuon` in `crates/nu-command/src/formats/to/nuon.rs`
- the signature now features `--raw`, `--indent <int>` and `--tabs <int>`
- the structure of the `run` method is inspired from the one in `to json`
  - we get the values of the arguments
  - we convert the input to a usable `Value`
  - depending on whether the user raised `--raw`, `--indent` or `--tabs`, we call the conversion to `string` with different values of the indentation, starting at depth 0
  - finally, we return `Ok` or a `ShellError::CantConvert` depending on the conversion result
- some tool functions
  - `get_true_indentation` gives the full indentation => `indent` repeated `depth` times
  - `get_true_separators` gives the line and field separators => a `("\n", "")` when using some formatting or `("", " ")` when converting as pure string on a single line 

the meat of `nuon.rs` is now the `value_to_string` recursive function:
- takes the depth and the indent string
- adds correct newlines, space separators and indentation to the output
- calls itself with the same indent string but `depth + 1` to increase the indentation by one level
- i used the `nl`, `idt`, `idt_po` (**i**n**d**en**t** **p**lus **o**ne) and `idt_pt` (**i**n**d**en**t** **p**lus **t**wo) to make the `format!`s easier to read

# User-Facing Changes
users can now
- control the amount and nature of NUON string output indentation with
  - `--indent <number of " " per level>`
  - `--tabs <number of "\t" per level>` 
- use the previous behaviour of `to nuon` with the `--raw` option
- have new examples with `help to nuon`

> **Note**
> the priority order of the options is the following
> 1. `--raw`
> 2. `--tabs`
> 3. `--indent`
>
> the default is `--indent 2`

# Tests + Formatting
### new tests
- tests involving the string output of `to nuon`, i.e. tests not of the form `... | to nuon | from nuon ...`, now use the `to nuon --raw` command => this is the smallest change to have the tests pass, as the new `to nuon --raw` is equivalent to the old `to nuon`
- in `crates/nu-command/src/formats/to/nuon.rs`, the previous example has been replaced with three examples
  - `[1 2 3] | to nuon` to show the default behaviour
  - `[1 2 3] | to nuon --raw` to show the not-formatted output
  - a more complex example with `{date: 2000-01-01, data: [1 [2 3] 4.56]} | to nuon`
  - the result values have been defined and the `examples` tests pass
 
### dev
- :green_circle: `cargo fmt --all`
- :green_circle: `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect`
- :green_circle: `cargo test --workspace` ~~passes but without `to_nuon_errs_on_closure`~~ fixed in 0b4fad7effd5a3adf0ccf2aa694d7a77653f1b55

# After Submitting
the `to nuon` page would have to be regenerated at some point due to the new tests

### CODE REVIEW

**Comment by amtoine** [comment]
hello there :wave: :yum: 

i think this PR is almost ready
- the feature is there
- almost all the tests pass

this is where i have a question for ya :smirk: 

in b151294ab, i've disabled the two tests that do not pass
- `to_nuon_range` => this is a bug that i have to address
- `to_nuon_errs_on_closure` => i think this is a design question

before the PR, `{||} | to nuon` would return a `nu::shell::unsupported_input` error.
now, it returns a `nu::shell::cant_convert` error, and thus the `to_nuon_errs_on_closure` does not pass :thinking: 

:one: should i keep the previous `nu::shell::unsupported_input` error by applying this?
```diff
diff --git a/crates/nu-command/src/formats/to/nuon.rs b/crates/nu-command/src/formats/to/nuon.rs
index db64119a5..0d3652659 100644
--- a/crates/nu-command/src/formats/to/nuon.rs
+++ b/crates/nu-command/src/formats/to/nuon.rs
@@ -73,15 +73,7 @@ impl Command for ToNuon {
                 span,
             }
             .into_pipeline_data()),
-            _ => Ok(Value::Error {
-                error: ShellError::CantConvert {
-                    to_type: "NUON".into(),
-                    from_type: value.get_type().to_string(),
-                    span,
-                    help: None,
-                },
-            }
-            .into_pipeline_data()),
+            Err(error) => Err(error),
         }
     }
 
diff --git a/crates/nu-command/tests/format_conversions/nuon.rs b/crates/nu-command/tests/format_conversions/nuon.rs
index 2c4d6bfeb..cd86b864e 100644
--- a/crates/nu-command/tests/format_conversions/nuon.rs
+++ b/crates/nu-command/tests/format_conversions/nuon.rs
@@ -285,7 +285,7 @@ fn from_nuon_datetime() {
     assert_eq!(actual.out, "date");
 }
 
-#[allow(dead_code)]
+#[test]
 fn to_nuon_errs_on_closure() {
     let actual = nu!(
         cwd: "tests/fixtures/formats", pipeline(
```
:two: should i change the test to switch to the `to_nuon_errs_on_closure` error with the following changes?
```diff
diff --git a/crates/nu-command/tests/format_conversions/nuon.rs b/crates/nu-command/tests/format_conversions/nuon.rs
index 2c4d6bfeb..f397a89e2 100644
--- a/crates/nu-command/tests/format_conversions/nuon.rs
+++ b/crates/nu-command/tests/format_conversions/nuon.rs
@@ -295,7 +295,7 @@ fn to_nuon_errs_on_closure() {
         "#
     ));
 
-    assert!(actual.err.contains("not nuon-compatible"));
+    assert!(actual.err.contains("can't convert closure to NUON"));
 }
 
 #[test]
```

> **Note**
> the two solutions allow `cargo test --workspace` to pass :ok_hand:
> so i do not mind choosing one or the other, your call :wink:

**Comment by amtoine** [comment]
i'm starting working on the `to_nuon_range` problem :ok_hand:

**Comment by amtoine** [comment]
tests fixed in 0b4fad7effd5a3adf0ccf2aa694d7a77653f1b55 :ok_hand:

**Comment by amtoine** [comment]
in the four latest commits, i've made "raw" the default behaviour.

the motivation comes from a remark of @fdncred i found very sensible!
> I think one of the big selling points of `nuon` is how it's minified.
> [...]
> I figure we'd want the `--raw` output as the default and then `--pretty` if they don't want raw, for nuon

> **Note**
> see the [original discussion](https://discord.com/channels/601130461678272522/615962413203718156/1086365540454518784)

in the end i did not implement a `--pretty` option, but rather kept the `--indent` and `--tabs` to play this role
- `to nuon` will use the "raw" format, the new default
- `to nuon --indent i`, `to nuon --tabs t` and `to nuon --indent i --tabs t` will use the new "pretty" format
- `to nuon --raw` and any `to nuon {flags} --raw` will use the "raw" option, disabling all the pretty output

## changelog
- 6bc95ce0a makes the "raw" format the default behaviour again, disabling the pretty ones
- 86261f55f fixes the documentation and the examples
- a82360047 reverts the `--raw` option change i've made to all the tests in 9159377af => this makes the overall PR change muuuch smaller

:test_tube: all the tests pass locally

**Comment by fdncred** [comment]
Seems like a good thing. Thanks!

**Comment by amtoine** [comment]
> Seems like a good thing. Thanks!
>
yeaah :partying_face:
